### PR TITLE
[RSDK-8807] support reading array of ints from config in C FFI

### PR DESCRIPTION
**Pull request by gvaradarajan**
Includes support for building the FFI with pre-populated robot credentials

### CODE REVIEW

File: micro-rdk-ffi/example/example.c [hunk 1]
```diff
@@ -24,9 +25,38 @@ int config_my_generic_sensor_A(struct config_context *ctx, void *user_data,
   struct my_generic_sensor_A *sensorA = malloc(sizeof(struct my_generic_sensor_A));
 
   sensorA->an_int = 1234567;
-  sensorA->array =  NULL;
   sensorA->an_int_from_config = my_int;
-  
+  int32_t vec_len = -1;
+  viam_code ret2 = config_get_i32_vec_len(ctx, "my_list", &vec_len);
+  if (ret2 != VIAM_OK) {
```
Comment (npmenard) [inline micro-rdk-ffi/example/example.c:31]: If we can sanitize that code a bit so it's more readable that would be great!!
can return early when running into an error

File: micro-rdk-ffi/src/ffi/config.rs [hunk 2]
```diff
@@ -135,3 +135,70 @@ pub unsafe extern "C" fn config_get_i32(
...
+/// if found the values will be copied into `out`. The length should be obtained
+/// first, using `config_get_i32_vec_len` in order to allocate the proper amount of memory
+/// pointed to by `out`
+///
+/// # Safety
+/// `ctx`, `key`, `out` must be valid pointers for the duration of the call
+/// `key` must be a null terminated C string. Additionally the external process calling
+/// the function is responsible for managing the memory allocated for `out`
+#[no_mangle]
+pub unsafe extern "C" fn config_get_i32_vec(
```
Comment (npmenard) [inline micro-rdk-ffi/src/ffi/config.rs:149]: Did you consider returning a rust allocated array instead? so be copied in C and returned or kept until not needed anymore?

**Review by npmenard — APPROVED** [review]
LGTM, left a couple of comment.

File: micro-rdk-ffi/src/ffi/config.rs (hunk 2 above)
Comment (gvaradarajan) [reply micro-rdk-ffi/src/ffi/config.rs:149]: I did but I thought having the Rust code write to memory pre-allocated by C was safer and simpler? I might not have approached the Rust allocated array properly when I attempted it

File: micro-rdk-ffi/src/ffi/runtime.rs [hunk 3]
```diff
@@ -263,12 +263,21 @@ pub unsafe extern "C" fn viam_server_start(ctx: *mut viam_server_context) -> via
...
-            ROBOT_SECRET.expect("Provided build-time configuration failed to set `ROBOT_SECRET`"),
-        );
+        use micro_rdk::common::credentials_storage::RobotConfigurationStorage;
+        use micro_rdk::proto::provisioning::v1::CloudConfig;
+
+        let mut ram_storage = RAMStorage::new();
+        // TODO: [RSDK-8923] Get app_address from machine credentials
+        let cloud_conf = CloudConfig {
+            id: ROBOT_ID
+                .expect("Provided build-time configuration failed to set `ROBOT_ID`")
```
[the remark this answers is not in the document]
Comment (gvaradarajan) [reply micro-rdk-ffi/src/ffi/runtime.rs:273]: will do

File: micro-rdk-ffi/example/example.c (hunk 1 above)
Comment (gvaradarajan) [reply micro-rdk-ffi/example/example.c]: I adjusted it and cleaned it up a little, was it what you had in mind?

**Pull request merged**